jbpm-flow/src/main/java/org/jbpm/workflow/instance/impl/WorkflowProcessInstanceImpl.java

```diff
@@ -654,7 +657,30 @@ public void signalEvent(String type, Object event) {
 		}
 	}
 
-	private void validate() {
+	private Function<String, String> getResolver(Node node, String type, List<NodeInstance> currentView) {
+        if (node instanceof DynamicNode) {
+            // special handling for dynamic node to allow to resolve variables from individual node instances of the dynamic node
+            // instead of just relying on process instance's variables
+            return (e) -> {
+                List<NodeInstance> nodeInstances = getNodeInstances(node.getId(), currentView);
+                if (nodeInstances != null && !nodeInstances.isEmpty()) {
+                    StringBuilder st = new StringBuilder();
+                    for (NodeInstance ni : nodeInstances) {
+                        String result = resolveVariable(e, new NodeInstanceResolverFactory(ni));
+                        st.append(result).append("###");
+                    }
+                    return st.toString();
```

Comment:
it is used by DynamicNode.acceptEvent that is checking based on contains method instead of equals. It's just a check on definition level if there is any possible node instance that would accept it and if so let it signal them

Reply:
ok, but it seems to me that this resolver is only applied to the subnodes of adHocSubProcess/stage and not the adHocSubProcess/stage itself. Is that right?

Reply:
```
List<NodeInstance> nodeInstances = getNodeInstances(node.getId(), currentView);
```
will collect all node instances (currently active) for given node id, and the node is is the adhoc/stage node id here

Reply:
yes, but in DynamicNode.acceptEvent it is passing subnodes names as variables to resolve, right? 
```
for (Node node : getNodes()) {
            if (resolver.apply(node.getName()).contains(type) && node.getIncomingConnections().isEmpty()) {
                return true;
            }
        }
```

Reply:
yes, because it checks if there is anything that can accept given signal, regardless of which node as this is preliminary verification.

Reply:
thanks, @mswiderski!